rspace/src/main/scala/coop/rchain/rspace/serializers/ScodecSerialize.scala

```diff
@@ -216,7 +195,31 @@ object ScodecSerialize {
         case p: NodePointer => p
       }(Blake2b256Hash.codecWithBytesStringBlake2b256Hash.as[NodePointer])
 
-  implicit def codecTrieValuePointer: Codec[ValuePointer] =
+  /**
+    * FIXME There some problem with fixed length vector for `VectorOfN(provide(2), int32)` which when you encode
+    *        Vector(1) with only one element would be successful but decode the generated byteVector would fail.
+    *        example:
+    *         val codecVector = vectorOfN(provide(2), int32)
+    *         val encoded = codecVector.encode(Vector(1))  // success
+    *         val decoded = codecVector.decode(encoded.toOption.get)  // fail
+    *        There are limitation on creating method of PointerBlock which garantees length 256.
```

Comment:
Maybe I should fix this now?

Reply:
This is expected behavior with `provide` codec because information about length is not part of encoded data, so decoding the value which is not the same length as specified in _provide_ will fail.

Here is the example, using _provide_ will not encode length so decoder will expect the size 3.
```scala
val c1 = vectorOfN(provide(3), uint4)
c1.encode(Vector(1, 2, 3)).getUnsafe // BitVector(12 bits, 0x123)

val c2 = vectorOfN(uint4, uint4)
c2.encode(Vector(1, 2, 3)).getUnsafe // BitVector(16 bits, 0x3123)
```

Because PointerBlock is always created with fixed length (256), _provide_ codec can be used here.
https://github.com/rchain/rchain/blob/02c276ff4fe04415b0262e02fb3dd69ebda48533/rspace/src/main/scala/coop/rchain/rspace/history/History.scala#L120-L123

Reply:
This is an unsafe function from my perspective. Anyway, just want to mention that.

Reply:
I would remove the comment.